### PR TITLE
feat: add support for custom wakatime api domain

### LINKED ISSUE #608: Make Wakatime endpoint URL configurable
I'd love to have the ability to configure the Wakatime API endpoint via a config file and / or environment variable. This way, alternative Wakatime-compatible backends like [Wakapi](https://github.com/muety/wakapi/issues/65) could easily be plugged in as well.

What do you think?

### PR DESCRIPTION
**Pull request by nathanchu**
Fixes #608

Demo (https://github-readme-stats-git-wakatime-custom-url.nathanchu.vercel.app/api/wakatime?username=nathanchu&api_domain=wakatime.com):
[](https://github-readme-stats-git-wakatime-custom-url.nathanchu.vercel.app/api/wakatime?username=nathanchu&api_domain=wakatime.com)

### CODE REVIEW

**Review by qwerty541 — COMMENTED** [review]
I think you also should add info about this new option into the docs

File: src/fetchers/wakatime-fetcher.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
 const axios = require("axios");
 
-const fetchLast7Days = async ({ username }) => {
+const fetchLast7Days = async ({ username, api_domain }) => {
   try {
     const { data } = await axios.get(
-      `https://wakatime.com/api/v1/users/${username}/stats/last_7_days?is_including_today=true`,
+      `https://${
```
Comment (anuraghazra) [inline src/fetchers/wakatime-fetcher.js:6]: Hopefully there won't be any XSS issues because of this. we will sanitize these interpolations in future.

Comment (nathanchu) [reply src/fetchers/wakatime-fetcher.js:6]: I don't know how that should be encoded, maybe strip all non "domain safe" characters (regex `/[^a-z-.0-9]/gi` i think)

Comment (anuraghazra) [reply src/fetchers/wakatime-fetcher.js:6]: That might work for now.

File: src/fetchers/wakatime-fetcher.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+        api_domain || "wakatime.com"
```
Comment (nathanchu) [inline src/fetchers/wakatime-fetcher.js:7]:
```suggestion
        api_domain && api_domain.replace(/[^a-z-.0-9]/gi, "") || "wakatime.com"
```

File: src/fetchers/wakatime-fetcher.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const fetchLast7Days = async ({ username, api_domain }) => {
   try {
     const { data } = await axios.get(
       `https://${
-        api_domain || "wakatime.com"
+        api_domain && api_domain.replace(/[^a-z-.0-9]/gi, "") || "wakatime.com"
```
Comment (ensia96) [inline src/fetchers/wakatime-fetcher.js:7]:
```suggestion
        api_domain?.replace(/[^a-z-.0-9]/gi, "") || "wakatime.com"
```

I think the code change you suggested can be organized in this way as well. : )

Or, like this

```suggestion
        api_domain ? api_domain.replace(/[^a-z-.0-9]/gi, "") : "wakatime.com"
```

**Comment by muety** [comment]
What has happened to this PR? I'd really like to have that feature! :)

**Comment by anuraghazra** [comment]
> What has happened to this PR? I'd really like to have that feature! :)

Sorry it got stale., totally lost the track of it. I'll merge this now.

**Pull request merged**